src/supertux/game_object_manager.cpp

```diff
@@ -376,79 +379,233 @@ GameObjectManager::on_editor_save()
   m_last_saved_change = (m_undo_stack.empty() ? UID() : m_undo_stack.back().uid);
 }
 
+void
+GameObjectManager::apply_object_change(const GameObjectChange& change, bool track_undo)
+{
+  GameObject* object = get_object_by_uid<GameObject>(change.uid);
+  switch (change.action)
+  {
+    case GameObjectChange::Action::CREATE:
+    {
+      create_object_from_change(change, track_undo);
+    }
+    break;
+
+    case GameObjectChange::Action::DELETE:
+    {
+      if (!object)
+        throw std::runtime_error("Object '" + change.name + "' does not exist.");
+
+      object->m_track_undo = track_undo;
+      object->remove_me();
+    }
+    break;
+
+    case GameObjectChange::Action::MODIFY:
+    {
+      if (!object)
+        throw std::runtime_error("Object '" + change.name + "' does not exist.");
+
+      auto settings = object->get_settings();
+      if (track_undo)
+        settings.save_state();
+
+      parse_object_settings(settings, change.data); // Parse settings
+      object->after_editor_set();
+
+      if (track_undo)
+        save_object_change(*object, settings);
+    }
+    break;
+
+    default:
+      break;
+  }
+}
+
+void
+GameObjectManager::apply_object_changes(const GameObjectChanges& changes, bool track_undo)
+{
+  for (const auto& change : changes.objects)
```

Comment:
If you're iterating over the changed objects, shouldn't the variable be called `object`? But then line 433 doesn't make any sense.

Reply:
The `objects` member of `GameObjectChanges` actually is a `GameObjectChange` vector. So the problem here is more so the naming of that member (maybe I can name it `changes`).

Reply:
How about renaming the class to `GameObjectChangeSet`. Then you can rename `objects` to `changes` and keep the UID for identifying that excrement.